File: lib/models/plex_role.dart
import 'package:json_annotation/json_annotation.dart';

import '../utils/json_utils.dart';

part 'plex_role.g.dart';

@JsonSerializable()
class PlexRole {
  @JsonKey(fromJson: flexibleInt)
  final int? id;
  final String? filter;
  final String tag;
  final String? tagKey;
  final String? role;
  final String? thumb;
  @JsonKey(fromJson: flexibleInt)
  final int? count;

  PlexRole({this.id, this.filter, required this.tag, this.tagKey, this.role, this.thumb, this.count});

  factory PlexRole.fromJson(Map<String, dynamic> json) => _$PlexRoleFromJson(json);

  Map<String, dynamic> toJson() => _$PlexRoleToJson(this);
}

File: apple-tv/VibeStream/Models/PlexRole.swift
import Foundation

struct PlexRole: Codable, Identifiable, Hashable {
    var id: Int?
    var count: Int?
    var filter: String?
    var tagKey: String?
    var role: String?
    var thumb: String?
    let tag: String

    var displayName: String { tag }
    var displayRole: String? { role }
}
